### PR TITLE
Fix repl initialization with a different transport

### PR DESCRIPTION
**Pull request by pfeodrippe**
If the transport is set and it's not part of nREPL itself,
it has to be required before starting the server.

### CODE REVIEW

**Comment by pfeodrippe** [comment]
You will be able to use `lein repl :headless` with new transports.

`lein repl` calls `REPL-y` which does is not updated yet.

**Comment by bbatsov** [comment]
We already require and resolve the transport var here https://github.com/technomancy/leiningen/blob/master/src/leiningen/repl.clj#L44

I assume this is enough, right?

File: src/leiningen/repl.clj [hunk 1]
```diff
@@ -199,27 +199,29 @@
...
-      (when ~start-msg?
-        (println "nREPL server started on port" port# "on host" ~(:host cfg)
-                 (str "- "
-                      (nrepl.transport/uri-scheme ~(or (:transport cfg) #'nrepl.transport/bencode))
-                      "://" ~(:host cfg) ":" port#)))
-      (spit (doto repl-port-file# .deleteOnExit) port#)
-      (when legacy-repl-port#
-        (spit (doto legacy-repl-port# .deleteOnExit) port#))
-      @(promise))
+  [`(do (if ~(some-> (:transport cfg) meta :ns str)
```
Comment (bbatsov) [inline src/leiningen/repl.clj]: Keep in mind this is not nREPL's raw config, but something that has already been processed it went to Lein's `cfg` map.

File: src/leiningen/repl.clj [hunk 2]
```diff
@@ -252,11 +254,17 @@
      `(do (try (require '~(init-ns project)) (catch Exception t#))
           (require ~@(init-requires project 'reply.main))))))
 
-(def ack-server
+(defn ack-server
```
Comment (bbatsov) [inline src/leiningen/repl.clj]: Why do we care about the transport of the ack server at all? It simply has to acknowledge one message - I guess assuming it's using bencode is perfectly fine.

File: src/leiningen/repl.clj [hunk 3]
```diff
@@ -252,11 +254,17 @@
...
   "The server which handles ack replies."
-  (delay (nrepl.server/start-server
-          :bind (repl-host nil)
-          :handler (nrepl.ack/handle-ack nrepl.server/unknown-op))))
+  [transport]
+  (if transport
```
Comment (bbatsov) [inline src/leiningen/repl.clj]: Assuming we care about the transport - I think that passing `nil` will select `bencode`, which would simplify this code.

File: src/leiningen/repl.clj (hunk 1 above)
Comment (bbatsov) [reply src/leiningen/repl.clj]: Ah, yeah. I keep forgetting how Lein runs the plugins in a different process. I guess you should add some explanations for this for future reference. I'd also try to reduce this code duplication in the `some->` and the `require`.

**Comment by bbatsov** [comment]
Apart from my minor remark about the code duplication everything looks good. I guess you should also mention this in the changelog.

**Comment by technomancy** [comment]
Thanks for looking at this.

**Pull request merged**